src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -542,4 +585,57 @@ TokenRefresher create(FirebaseApp app) {
       STOPPED
     }
   }
+
+  private static class DefaultConfigOptions{
+    private String databaseURL;
+    private String storageBucket;
+    private Map<String, Object> databaseAuthVariableOverride;
+    private String projectId;
+
+    private DefaultConfigOptions() {}
+
+    private static DefaultConfigOptions getOptionsFromEnv() {
+      String defaultConfig = System.getenv(DEFAULT_CONFIG_OPTIONS_VAR);
+      if (defaultConfig == null || defaultConfig.isEmpty()) {
+        return new DefaultConfigOptions();
+      }
+      Gson gson = new Gson();
+      DefaultConfigOptions defaultConfigOptions;
+      if (defaultConfig.charAt(0) == '{') {
+        defaultConfigOptions = gson.fromJson(defaultConfig, DefaultConfigOptions.class);  
+      } else {
+        FileReader reader;
+        try {
+          reader = new FileReader(defaultConfig);
+        } catch (FileNotFoundException e) {
+          throw new IllegalStateException(e)  ;
+        }
+        defaultConfigOptions = gson.fromJson(reader, DefaultConfigOptions.class);
+      }
+      if (defaultConfigOptions == null) {
+        throw new IllegalStateException("null JSON");
+      }
+      return defaultConfigOptions;
+    }
+
+    private static FirebaseOptions.Builder getDefaultOptionsBuilder(
+        DefaultConfigOptions defaultConfigOptions) {
+      FirebaseOptions.Builder builder = new FirebaseOptions.Builder();
+      if (defaultConfigOptions.databaseAuthVariableOverride != null 
```

Comment:
Wish we could just unmarshal the JSON into FirebaseOptions here (like we did in Node). But that's probably not possible due to the use of builder pattern here. Oh well.